src/index.ts

```diff
@@ -66,6 +66,18 @@ export class Collection<K, V> extends Map<K, V> {
 		return super.clear();
 	}
 
+	/**
+	 * Gets an element if the key exists, otherwise sets it to {@param defaultValue} and returns the {@param defaultValue}.
+	 * @param {*} key - Key to get from/set to the collection.
+	 * @param {*} defaultValue - Default value to be set and returned if the key doesn't exist.
+	 * @returns {*}
+	 */
+	public ensure(key: K, defaultValue: V): V {
+		if (this.has(key)) return this.get(key)!;
```

Comment:
You can skip a second map scan:
```suggestion
		const value = this.get(key);
		if (typeof value !== 'undefined') return value;
```

Reply:
Mmh what if the the collection has that key but its value is `undefined`?

Reply:
I agree, I think an additional scan to guard against this is preferable over saving a couple of milliseconds and potentially setting the default instead of recognizing that the key already exists in the collection.